Mathlib/Analysis/SpecialFunctions/Pow/Continuity.lean

```diff
@@ -417,6 +417,19 @@ theorem continuousAt_rpow_const {x : ℝ≥0} {y : ℝ} (h : x ≠ 0 ∨ 0 ≤ y
 theorem continuous_rpow_const {y : ℝ} (h : 0 ≤ y) : Continuous fun x : ℝ≥0 => x ^ y :=
   continuous_iff_continuousAt.2 fun _ => continuousAt_rpow_const (Or.inr h)
 
+@[fun_prop]
+theorem continuousOn_rpow_const_compl_zero {r : ℝ} :
+    ContinuousOn (fun z : ℝ≥0 => z ^ r) {0}ᶜ :=
+  fun _ h => ContinuousAt.continuousWithinAt <| NNReal.continuousAt_rpow_const (.inl h)
+
+-- even though this follows from `ContinuousOn.mono` and the previous lemma, we include it for
+-- automation purposes with `fun_prop`, because the side goal `0 ∉ s ∨ 0 ≤ r` is often easy to check
+@[fun_prop]
+theorem continuousOn_rpow_const {r : ℝ} {s : Set ℝ≥0}
+    (h : 0 ∉ s ∨ 0 ≤ r) : ContinuousOn (fun z : ℝ≥0 => z ^ r) s :=
+  h.elim (fun _ ↦ ContinuousOn.mono (s := {0}ᶜ) (by fun_prop) (by aesop))
+    (NNReal.continuous_rpow_const · |>.continuousOn)
```

Comment:
So I found out something *very* unfortunate here. The stuff that was supposed to be fast from this is slow! :fearful: It turns out that I had declared this and the attribute inside the file we were working on, so it was tried first and was fast. But at this location, it is tried later, and it's very slow (and seemingly for no good reason if you look at the trace). We desperately need to figure that out. Sorry this PR is turning out to be so much work, but I think we're learning some valuable lessons here, and fixing this kind of stuff now is important before we get in too deep.

Reply:
By the way, you can tell it's just about the location where it's declared, because you can "redeclare" the attribute with `attribute [fun_prop] continuousOn_rpow_const` in the `CFC.rpow` file and the immediately become fast again.

Reply:
I'm starting to think that `fun_prop` isn't the problem at all, *everything* is slow in this file (and pretty much every other file about the CFC), especially in the last section where we use both the unital and non-unital CFC. Try sorry-ing out a bunch of lemmas and you'll see what I mean: even then it's slow!

Reply:
I think whatever is causing that problem is what is slowing down fun_prop: if it takes ~0.5s when it immediately picks the right lemma, it's not surprising that it takes 4s when it has to try 8 things.